src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,30 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateFromD = LocalDate.parse(dateFrom, DateTimeFormatter.ofPattern("dd.MM.yyyy"));
+        LocalDate dateToD = LocalDate.parse(dateTo, DateTimeFormatter.ofPattern("dd.MM.yyyy"));
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (String namesS : names) {
+            int salary = 0;
+            result.append(System.lineSeparator()).append(namesS).append(" - ");
+            for (String datum : data) {
+                String[] datePeriod = datum.split(" ");
```

Comment:
```suggestion
                String[] arrayOfData = datum.split(" ");
```
there will be not only dates in this array

Reply:
ok